python_tsp/heuristics/local_search.py

```diff
@@ -78,12 +78,19 @@ def solve_tsp_local_search(
         improvement = False
         for n_index, xn in enumerate(neighborhood_gen[perturbation_scheme](x)):
             if default_timer() - tic > max_processing_time:
-                logger.warning("Stopping early due to time constraints")
+                warning_msg = "Stopping early due to time constraints"
+                if log_file:
+                    logger.warning(warning_msg)
```

Comment:
So, instead of keeping the `logger` (which has been shown to be a major source of confusion when a root level file also set up a logger as we talked in the issue), can you replace it with a regular print but printing to a file?

It would be something like this. First, you open the file at the beginning of the execution:

```python
log_file_handler = open(log_file, "w", encoding="utf-8")
```

Then here you would call 
```suggestion
                    print(warning_msg, file=log_file_handler)
```

Then at the end of the execution before returning you close it

```python
log_file_handler.close()
```